src/librustc_borrowck/borrowck/gather_loans/mod.rs

```diff
@@ -201,6 +219,19 @@ fn check_aliasability<'a, 'tcx>(bccx: &BorrowckCtxt<'a, 'tcx>,
     }
 }
 
+fn check_yields<'a, 'tcx>(bccx: &BorrowckCtxt<'a, 'tcx>,
```

Comment:
I was looking around there and that doesn't seem correct. No loan records are created when we get `RestrictionResult::Safe`, so I assume that means the loans don't show up with `CheckLoans::each_in_scope_loan()` leading to unsoundness. We also want to ignore `ReEarlyBound` and `ReFree` borrows and it looks like that information is lost by then.